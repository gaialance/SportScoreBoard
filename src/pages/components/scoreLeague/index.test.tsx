import { render,screen , cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import '@testing-library/jest-dom/extend-expect'

import ScoreLeague, { sortScore } from './'


describe("rending the component",()=>{
    it("start the rending the first data",()=>{
        const { container } = render( <ScoreLeague leagueId={0}/>)

        expect(container).toMatchSnapshot();
    })
    

    it("testing the sort function for the score board",()=>{
        const testData = [
            {
                homeTeam:"Nottom Forest",
                awayTeam:"Newcastle",
                homeTeamScore:2,
                awayTeamScore:4,
                date:new Date('2023-03-18 04:00:00 +08:00'),
              },
              {
                homeTeam:"Aston Villa",
                awayTeam:"Bournemouth",
                homeTeamScore:5,
                awayTeamScore:4,
                date:new Date('2023-03-18 05:00:00 +08:00'),
              },
              {
                homeTeam:"BrentFord",
                awayTeam:"Leicester",
                homeTeamScore:5,
                awayTeamScore:4,
                date:new Date('2023-03-18 10:00:00 +08:00'),
              }
        ]

        const expectedData = [
            {
                homeTeam:"BrentFord",
                awayTeam:"Leicester",
                homeTeamScore:5,
                awayTeamScore:4,
                date:new Date('2023-03-18 10:00:00 +08:00'),
            },
            {
                homeTeam:"Aston Villa",
                awayTeam:"Bournemouth",
                homeTeamScore:5,
                awayTeamScore:4,
                date:new Date('2023-03-18 05:00:00 +08:00'),
            },
            {
                homeTeam:"Nottom Forest",
                awayTeam:"Newcastle",
                homeTeamScore:2,
                awayTeamScore:4,
                date:new Date('2023-03-18 04:00:00 +08:00'),
            }, 
        ]
        render( <ScoreLeague leagueId={0}/>)

        expect( screen.getByTestId("title") ).toHaveTextContent("Premier League")
        
        //Act
        const testOutput = sortScore( testData )


        //Assert output
        expect ( testOutput ).toEqual(expectedData)
    })
})

