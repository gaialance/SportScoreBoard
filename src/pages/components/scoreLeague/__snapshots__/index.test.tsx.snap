// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rending the component start the rending the first data 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="container MuiBox-root css-0"
    >
      <div
        class="MuiStack-root css-163ljxj-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          data-testid="title"
        >
          Premier League
        </h4>
        <div
          class="MuiStack-root css-1e7mpa5-MuiStack-root"
          data-testid="ScoreBoard-1"
        >
          <div
            class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-19otjfj-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeam"
                  >
                    BrentFord
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                  >
                    vs
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeam"
                  >
                    Leicester
                  </h5>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeamScore"
                  >
                    5
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                />
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeamScore"
                  >
                    4
                  </h5>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-19otjfj-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeam"
                  >
                    Aston Villa
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                  >
                    vs
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeam"
                  >
                    Bournemouth
                  </h5>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeamScore"
                  >
                    5
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                />
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeamScore"
                  >
                    4
                  </h5>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-19otjfj-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeam"
                  >
                    Nottom Forest
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                  >
                    vs
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeam"
                  >
                    Newcastle
                  </h5>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="homeTeamScore"
                  >
                    2
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
                />
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                    data-testid="awayTeamScore"
                  >
                    4
                  </h5>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
                  data-testid="buttonTest"
                  tabindex="0"
                  type="button"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Create New
                  </p>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
